src/Nether.Analytics.EventHubs/EventHubOutputManager.cs

```diff
@@ -2,28 +2,88 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 using System;
-using Nether.Analytics.Parsers;
 using System.Threading.Tasks;
+using Microsoft.Azure.EventHubs;
+using System.Text;
 
 namespace Nether.Analytics
 {
     public class EventHubOutputManager : IOutputManager
     {
-        private string _outputEventHubConnectionString;
+        private string _ehConnectionString;
+        private IOutputFormatter _serializer;
+        private EventHubClient _client;
+        private Encoding _encoding;
 
+        /// <summary>
+        /// Creates a new instance, with a JSON output formatter as the default serializer.
+        /// </summary>
+        /// <param name="outputEventHubConnectionString">The connection string for the event hub output.</param>
         public EventHubOutputManager(string outputEventHubConnectionString)
+            : this(outputEventHubConnectionString, new JsonOutputFormatter())
         {
-            _outputEventHubConnectionString = outputEventHubConnectionString;
+        }
+
+        public EventHubOutputManager(string outputEventHubConnectionString, IOutputFormatter serializer)
+            : this(outputEventHubConnectionString, serializer, Encoding.UTF8)
+        {
+        }
+
+        public EventHubOutputManager(string outputEventHubConnectionString, IOutputFormatter serializer, Encoding encoding)
+        {
+            _encoding = encoding;
+            _serializer = serializer;
+            _ehConnectionString = outputEventHubConnectionString;
+
+            // spin up the event hub client
+            EnsureEventHubClient();
```

Comment:
I like the EnsureEventHubClient pattern. In fact it's a pattern and Naming Convention we should adopt widely in Nether. I'm not too keen of running it in the constructor though since it kind of defeats the whole purpose of it. 

Another patterns that we've tried to followed is to have as few constructors as possible throw exceptions. Ensuring the clients could potentially throw exceptions here.

Please remove from constructor but keep otherwise.